src/mmw/js/src/core/modals/views.js

```diff
@@ -252,6 +254,58 @@ var MultiShareView = ModalBaseView.extend({
     onLinkToggle: function(e) {
         this.model.set('is_private', !e.target.checked);
         this.model.saveProjectAndScenarios();
+    },
+
+    connectHydroShare: function() {
+        var userHasHydroShareAccess = this.options.app.user.get('hydroshare');
+
+        if (userHasHydroShareAccess) {
+            // User already has connected their account. Allow them to turn
+            // this on and off at will.
+            // TODO Add project export ability.
+            return;
+        } else {
+            // User has not connectd to HydroShare yet. Have them sign in
+            // and allow MMW access, then enable the checkbox.
+
+            var self = this,
+                checkbox = self.ui.hydroShareEnabled,
+                confirm = new ConfirmLargeView({
+                        model: new models.ConfirmModel({
+                            titleText: 'Publish on HydroShare',
+                            question: [
+                                'You will need to connect your HydroShare ' +
+                                'account to your Model My Watershed account, ' +
+                                'or create a new HydroShare account.',
+
+                                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. ' +
+                                'Donec condimentum rutrum vulputate. ' +
+                                'Praesent id arcu id nisi elementum pretium at ut felis.'
+                            ],
+                            confirmLabel: 'Connect',
+                            cancelLabel: 'Cancel',
+                        })
+                    }),
+                iframe = new IframeView({
+                    model: new models.IframeModel({
+                        href: '/user/hydroshare/login/',
+                        signalSuccess: 'mmw-hydroshare-success',
+                        signalFailure: 'mmw-hydroshare-failure',
+                    })
+                });
+
+            checkbox.prop('checked', false);
+            confirm.render();
+
+            confirm.on('confirmation', iframe.render);
+            iframe.on('success', function() {
+                // Fetch user again to save new HydroShare Access state
```

Comment:
The word "save" here in this comment is a little confusing. Does calling `fetch()` persist something?

Reply:
When the OAuth dance succeeds, there is a new entry in the `user_hydrosharetoken` table for this user. When we call `fetch` again, the value of `user.hydroshare` is now `true`. This is the value that will be checked the next time the slider is slid.

I'm going to reword this and remove `save`.